### PR TITLE
Addon-docs: Render components as leaves in `--docs` mode

### PR DESCRIPTION
**Pull request by shilman**
Issue: #6646 

## What I did

Storybook_and_Comparing_next___6646-docs-mode-nav-ui_·_storybookjs_storybook

When user runs or builds using `--docs`, the component nodes in the nav hierarchy are collapsed into leaves (i.e. individual stories are not shown in the nav)

There's a corner case where a nav node contains both stories and sub-nodes. In this case the sub-nodes are removed and the user is warned.

If this behavior looks good:
- [x] Fix tree nav selection highlighting
- [x] Fix keyboard nav so that next/previous story actually go next/previous component

## How to test

```
cd examples/official-storybook
yarn storybook --docs
```

### CODE REVIEW

File: lib/ui/src/containers/nav.js [hunk 1]
```diff
@@ -95,6 +97,41 @@ const createMenu = memoize(1)((api, shortcutKeys, isFullscreen, showPanel, showN
...
+export const removeLeafStories = stories => {
+  return Object.values(stories).reduce((acc, item) => {
+    // filter out leaves
+    if (item.isLeaf) {
+      return acc;
+    }
+
+    const { children, ...rest } = item;
+    const component = { ...rest, isLeaf: true, isComponent: true, id: children[0] };
+    // make components leaves
```
Comment (tmeasday) [inline lib/ui/src/containers/nav.js]: Can you expand this comment: "Make component leaves and link to the first story"

File: lib/ui/src/containers/nav.js [hunk 2]
```diff
@@ -95,6 +97,41 @@ const createMenu = memoize(1)((api, shortcutKeys, isFullscreen, showPanel, showN
...
+    if (item.isComponent) {
+      return Object.assign(acc, { [item.id]: component });
```
Comment (tmeasday) [inline lib/ui/src/containers/nav.js]: Is it weird that we end up with an entry in the `acc` who's key (`item.id`) is different to the `id` (`[children[0]`)?

File: lib/ui/src/containers/nav.js [hunk 3]
```diff
@@ -105,12 +142,16 @@ export const mapper = ({ state, api }) => {
     storiesConfigured,
   } = state;
 
+  const stories = docsMode ? memoize(1)(removeLeafStories)(storiesHash) : storiesHash;
+
+  console.log({ stories });
```
Comment (tmeasday) [inline lib/ui/src/containers/nav.js]: Log?

**Review by tmeasday — COMMENTED** [review]
Seems OK, I'm not sure if the `id !== key` thing could bite us in other places though

**Comment by shilman** [comment]
@tmeasday updated & seems to be working

**Comment by joeycozza** [comment]
At my company, we are messing around and trying to figure out how we are going to utilize Docs.

We like the idea of utilizing a pattern where the .mdx file uses something like
```jsx
<Description of="." />
<Story id="." />
<Props of="." />

## Freeform Notes
```
 That way the currently selected story is being shown in the Documentation, and each story could have a different description/usage guide.

If we go that route, should we just not build with the `--docs` flag? If we do, it seems we lose the ability to have a per-story overriding of the Docs tab.

I wonder if our use case is common enough that it might be beneficial to have the option to NOT render components as leaves while building with `--docs`?

**Comment by shilman** [comment]
@joeycozza This is great feedback. I'll chew on it!

**Comment by matthewmatician** [comment]
This would be an awesome feature to have...

Right now I've got a designer who wants to write some docs for our components, and it'd be super great to have this ability to have a docs-only view of storybook.

In addition to this build, could we also have a docs-only view within storybook that you could turn on and off to get a preview of what the `--docs` output would look like?

Awesome stuff, keep it up!

**Comment by kevinSuttle** [comment]
This would be awesome. ➕ _1_

File: docs/src/new-components/basics/tooltip/TooltipLinkList.js [hunk 4]
```diff
@@ -7,7 +7,7 @@ const List = styled.div`
   min-width: 180px;
   overflow: hidden;
   overflow-y: auto;
-  maxHeight: ${10.5 * 32 /* 10.5 items */};
+  maxheight: ${10.5 * 32 /* 10.5 items */};
```
[the remark this answers is not in the document]
Comment (monospaced) [reply docs/src/new-components/basics/tooltip/TooltipLinkList.js]: Still missing the `-`?

**Comment by ndelangen** [comment]
@shilman what's the status on this?

**Comment by shilman** [comment]
@ndelangen I plan to pick it up after finishing higher priority work

**Comment by zachintosh** [comment]
Agree with the sentiment above -- this would be awesome to have.

**Comment by shilman** [comment]
Self-merging @tmeasday after UI approval from @domyen

**Pull request merged**